src/plugins/intel_cpu/tests/functional/single_layer_tests/shapeof.cpp

```diff
@@ -1,175 +1,201 @@
+
+#include "ngraph_functions/builders.hpp"
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+
+using InputShape = ov::test::InputShape;
+using ElementType = ov::element::Type_t;
+
+namespace CPULayerTestsDefinitions {
+typedef std::tuple<
+        InputShape,
+        ElementType                // Net precision
+> ShapeOfLayerTestParams;
+
+typedef std::tuple<
+        ShapeOfLayerTestParams,
+        CPUSpecificParams
+> ShapeOfLayerCPUTestParamsSet;
+
+class ShapeOfLayerCPUTest : public testing::WithParamInterface<ShapeOfLayerCPUTestParamsSet>,
+                            virtual public ov::test::SubgraphBaseTest, public CPUTestsBase {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<ShapeOfLayerCPUTestParamsSet> obj) {
+        CPULayerTestsDefinitions::ShapeOfLayerTestParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = obj.param;
+        ElementType netPr;
+        InputShape inputShape;
+
+        std::tie(inputShape, netPr) = basicParamsSet;
+        std::ostringstream result;
+        result << "ShapeOfTest_";
+        result << std::to_string(obj.index) << "_";
+        result << "Prec=" << netPr << "_";
+        result << CPUTestsBase::getTestCaseName(cpuParams) << "_";
+        result << "IS=";
+        result << CommonTestUtils::partialShape2str({inputShape.first}) << "_";
+        result << "TS=(";
+        for (const auto& shape : inputShape.second) {
+            result << CommonTestUtils::vec2str(shape) << "_";
+        }
+        result << ")";
+        return result.str();
+    }
+protected:
+    void SetUp() override {
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+
+        CPULayerTestsDefinitions::ShapeOfLayerTestParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+
+        auto netPrecision = ElementType::undefined;
+        InputShape inputShape;
+        std::tie(inputShape, netPrecision) = basicParamsSet;
+        init_input_shapes({inputShape});
+
+        inType = ov::element::Type(netPrecision);
+        outType = ElementType::i32;
+        selectedType = makeSelectedTypeStr("ref", inType);
+
+        auto params = ngraph::builder::makeDynamicParams(inType, inputDynamicShapes);
+        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::opset3::Parameter>(params));
+        auto shapeOf = std::make_shared<ngraph::opset3::ShapeOf>(paramOuts[0], ngraph::element::i32);
+        shapeOf->get_rt_info() = getCPUInfo();
+
+        const ngraph::ResultVector results{std::make_shared<ngraph::opset3::Result>(shapeOf)};
+        function = std::make_shared<ngraph::Function>(results, params, "ShapeOf");
```

Comment:
Could you please use `makeNgraphFunction` call, to properly initialize the runtime data?

Reply:
Yes; done